clang/test/SemaCXX/warn-unsafe-buffer-usage-fixits-add-assign.cpp

```diff
@@ -0,0 +1,55 @@
+// RUN: %clang_cc1 -std=c++20 -Wunsafe-buffer-usage \
+// RUN:            -fsafe-buffer-usage-suggestions \
+// RUN:            -fdiagnostics-parseable-fixits %s 2>&1 | FileCheck %s
+void foo(int * , int *);
+
+void add_assign_test(unsigned int n, int *a, int y) {
+  int *p = new int[10];
+  // CHECK: fix-it:"{{.*}}":{[[@LINE-1]]:3-[[@LINE-1]]:11}:"std::span<int> p"
+  // CHECK: fix-it:"{{.*}}":{[[@LINE-2]]:12-[[@LINE-2]]:12}:"{"
+  // CHECK: fix-it:"{{.*}}":{[[@LINE-3]]:23-[[@LINE-3]]:23}:", 10}"
+  p += 2;
+  // CHECK: fix-it:"{{.*}}":{[[@LINE-1]]:3-[[@LINE-1]]:9}:"p = p.subspan(2)"
```

Comment:
I suspect it needs to be like `p.subspan(0, 2)`. The first argument is offset, the second argument is count. (I wrote it weirdly on the wideboard many times because I never memorized any of this.)

Reply:
IIUC, the default for count is `std::dynamic_extent` which ensures the correct number of elements in the subspan (and I guess the fixit for UPC pre-increment makes this assumption as well.) Is this not the case?

Reply:
Wait never mind, offset is exactly what we need lol. I'm the one who got confused about what the intended behavior is, was thinking about the scenario where we're forcing a bounds check before calling `.data()` which is completely unrelated. Everything looks correct: https://godbolt.org/z/MWGEj4cT8